### PR TITLE
`Range`, `RDFT`, `RandormUniform`, `ReduceOps` layer tests to API2.0

### PR DESCRIPTION
**Pull request by vurusovs**

### CODE REVIEW

File: src/tests/functional/shared_test_classes/include/shared_test_classes/base/utils/ranges.hpp [hunk 1]
```diff
@@ -117,7 +117,7 @@ static std::map<ov::NodeTypeInfo, std::vector<std::vector<InputGenerateData>>> i
         { ov::op::v1::FloorMod::get_type_info_static(), {{{2, 4}}, {{2, 2, 128}}} },
         { ov::op::v1::Mod::get_type_info_static(), {{{2, 4}}, {{2, 2, 128}}} },
         { ov::op::v1::ReduceMax::get_type_info_static(), {{{0, 5}}, {{-5, 5, 1000}}} },
-        { ov::op::v1::ReduceMean::get_type_info_static(), {{{0, 5}}, {{0, 5, 1000}}} },
+        { ov::op::v1::ReduceMean::get_type_info_static(), {{{0, 5, 1000}}, {{0, 5, 1000}}} },
```
Comment (vurusovs) [inline src/tests/functional/shared_test_classes/include/shared_test_classes/base/utils/ranges.hpp:120]: @iefode without the change `ReduceOpsLayerTest` with `Mean` op type fails on integral types

**Review by iefode — APPROVED** [review]

File: src/tests/functional/shared_test_classes/include/shared_test_classes/base/utils/ranges.hpp (hunk 1 above)
Comment (vurusovs) [reply src/tests/functional/shared_test_classes/include/shared_test_classes/base/utils/ranges.hpp:120]: @iefode `0, 5, 1000` for all types (real and integral) were in original test. With `0, 5` only integral cases are failing in new tests

File: src/tests/functional/shared_test_classes/src/single_op/reduce_ops.cpp [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        default:
+            FAIL() << "Reduce op doesn't support operation type: " << op_type;
+    }
+    auto reduction_axes_node = std::make_shared<ov::op::v0::Constant>(ov::element::i64, ov::Shape(shape_axes), axes);
+
+    const auto reduce = ngraph::builder::makeReduce(param, reduction_axes_node, keep_dims, reduction_type);
+    function = std::make_shared<ov::Model>(reduce->outputs(), ov::ParameterVector{param}, "Reduce");
+}
+
+void ReduceOpsLayerWithSpecificInputTest::generate_inputs(const std::vector<ov::Shape>& targetInputStaticShapes) {
```
[the remark this answers is not in the document]
Comment (vurusovs) [reply src/tests/functional/shared_test_classes/src/single_op/reduce_ops.cpp:62]: No need, there is some specific use case (not for common use in other tests), so I ported it same way

**Comment by vurusovs** [comment]
`ie_tests_cpu_macos_arm64_release` has passed

**Pull request merged**